superduperdb/base/datalayer.py

```diff
@@ -803,7 +803,11 @@ def _add(
         if serialized is None:
             leaves = object.dict().get_leaves()
             leaves = leaves.values()
-            artifacts = [leaf for leaf in leaves if isinstance(leaf, Encodable)]
+            artifacts = [
+                leaf
+                for leaf in leaves
+                if isinstance(leaf, _BaseEncodable) and leaf.artifact
```

Comment:
_BaseEncodable 
self.artifact = True ? wdyt?

Reply:
We have a tag for this to tag the output of Encodable whether a artifact

Reply:
These are great points, we should handle all of this through the `db.load` API.